app/controllers/decidim/proposals/proposals_controller.rb

```diff
@@ -6,41 +6,49 @@ module Proposals
     class ProposalsController < Decidim::Proposals::ApplicationController
       helper Decidim::WidgetUrlsHelper
       helper ProposalWizardHelper
+      helper ParticipatoryTextsHelper
+      include Decidim::ApplicationHelper
       include FormFactory
       include FilterResource
       include Orderable
       include Paginable
 
-      helper_method :most_voted_positive_comment, :most_voted_negative_comment, :comment_author, :more_positive_comment, :get_comment, :comment_author_avatar, :get_positive_count_comment, :get_negative_count_comment, :process_categories, :has_categories, :get_id, :process_name, :my_category
+      helper_method :form_presenter, :most_voted_positive_comment, :most_voted_negative_comment, :comment_author, :more_positive_comment, :get_comment, :comment_author_avatar, :get_positive_count_comment, :get_negative_count_comment, :process_categories, :has_categories, :get_id, :process_name, :my_category
+
       before_action :authenticate_user!, only: [:new, :create, :complete]
       before_action :ensure_is_draft, only: [:compare, :complete, :preview, :publish, :edit_draft, :update_draft, :destroy_draft]
       before_action :set_proposal, only: [:show, :edit, :update, :withdraw]
       before_action :edit_form, only: [:edit_draft, :edit]
 
+      before_action :set_participatory_text
+
       def index
-        @proposals = search
-                     .results
-                     .published
-                     .not_hidden
-                     .includes(:category)
-                     .includes(:scope)
-
-        @voted_proposals = if current_user
-                             ProposalVote.where(
-                               author: current_user,
-                               proposal: @proposals.pluck(:id)
-                             ).pluck(:decidim_proposal_id)
-                           else
-                             []
-                           end
-
-        @proposals = paginate(@proposals)
-        @proposals = reorder(@proposals)
-
-        if params.has_key?(:filter)
-          @category_id = params[:filter][:category_id]
```

Comment:
Why you remove this?

Reply:
I updated the proposal controller to the [decidim 0.16 version](https://github.com/decidim/decidim/blob/d587a420e0f6c43aec7ed65a16f4b6f666bdac5c/decidim-proposals/app/controllers/decidim/proposals/proposals_controller.rb#L25) while keeping the custom functionality of best-comments.